### PR TITLE
UpSampling1D UpSampling2D

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

UpSampling1D UpSampling2D

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 1]
```diff
@@ -1592,3 +1592,181 @@ output m is : [[[[[ 0.24059629  0.11875484 -0.07601731  0.18490529]
...
+        [ 16.,  17.,  18.],
+        [ 16.,  17.,  18.]]], dtype=float32)
+```
+
+---
+## Upsampling2D ##
+
+**Scala:**
+```scala
+val module = UpSampling2D(size: (Int, Int), format: DataFormat = DataFormat.NCHW)
```
Comment (SeaOfOcean) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: Array[Int]?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/UpSampling2D.scala [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+ * If input's dataformat is NCHW, then the size of output is (N, C, H * size(0), W * size(1))
+ *
+ * @param size tuple of 2 integers. The upsampling factors for heights and widths.
+ * @param format DataFormat, NCHW or NHWC
+ * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
+ */
+class UpSampling2D[T: ClassTag] (val size: Array[Int], val format: DataFormat = DataFormat.NCHW)
+  (implicit ev: TensorNumeric[T]) extends TensorModule[T] {
+  require(size.length == 2, s"UpSampling2D's size should be an array containing" +
+    s" 2 elements, but got $size")
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/UpSampling2D.scala]: ${size.mkString("x")}

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 3]
```diff
@@ -1592,6 +1592,252 @@ output m is : [[[[[ 0.24059629  0.11875484 -0.07601731  0.18490529]
...
+```
+
+>>>>>>> Update Convolution-Layers.md
+
+**Python example:**
+```python
+from bigdl.nn.layer import *
+import numpy as np
+
+<<<<<<< 9cdea9becf90df1739bcb8d86716d7823c10e389
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: Remove this one?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 4]
```diff
@@ -1592,6 +1592,252 @@ output m is : [[[[[ 0.24059629  0.11875484 -0.07601731  0.18490529]
...
+    [ 0.52572423  0.52572423  0.04730832  0.04730832]
+    [ 0.74973559  0.74973559  0.2245238   0.2245238 ]
+    [ 0.74973559  0.74973559  0.2245238   0.2245238 ]]]]]
+```
+=======
+module = UpSampling1D(2)
+input = np.arange(1, 19).reshape(2, 3, 3)
+module.forward(input)
+```
+The output is
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: Seems that the layout has some problems.

**Comment by hkvision** [comment]
Can this be merged?

**Comment by wzhongyuan** [comment]
@hkvision   could you please run another PR job ? https://github.com/intel-analytics/BigDL/pull/2011 has refined the way how we add unit test for newly added layer to enable to auto-checking. seems you need to refine a little bit to follow it otherwise PR won't pass
thanks

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**